### PR TITLE
[improvement](disk balance) Prevent duplicate disk balance tasks afte…

### PR DESCRIPTION
**Pull request by deardeng**
…r a disk balance task has succ

## Proposed changes

1. When the BE side discovers that the migration path is already in the dest path, an error will be returned, and an OK should be returned at this time
2. After the task is completed, FE does not update the path hash of the replica current. This may still result in the tablet being disk rebalanced in the future. Until one minute later when the tablet is updated and reported

img_v2_136c7582-f2cc-414d-8457-2db5278270eg

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/agent/task_worker_pool.cpp [hunk 1]
```diff
@@ -2011,8 +2011,9 @@ Status StorageMediumMigrateTaskPool::_check_migrate_request(const TStorageMedium
         *dest_store = stores[0];
     }
     if (tablet->data_dir()->path() == (*dest_store)->path()) {
-        return Status::InternalError("tablet is already on specified path {}",
-                                     tablet->data_dir()->path());
+        LOG_WARNING("tablet is already on specified path")
```
[the remark this answers is not in the document]
Comment (deardeng) [reply be/src/agent/task_worker_pool.cpp:2014]: fixed

File: fe/fe-core/src/main/java/org/apache/doris/clone/TabletScheduler.java [hunk 2]
```diff
@@ -1642,6 +1656,7 @@ public boolean finishStorageMediaMigrationTask(StorageMediaMigrationTask migrati
             // if we have a success task, then stat must be refreshed before schedule a new task
             updateDiskBalanceLastSuccTime(tabletCtx.getSrcBackendId(), tabletCtx.getSrcPathHash());
             updateDiskBalanceLastSuccTime(tabletCtx.getDestBackendId(), tabletCtx.getDestPathHash());
+            updateDestPathHash(tabletCtx.getDestBackendId(), tabletCtx.getDestPathHash(), tabletCtx.getStorageMedium());
```
[the remark this answers is not in the document]
Comment (deardeng) [reply fe/fe-core/src/main/java/org/apache/doris/clone/TabletScheduler.java:1659]: fixed

**Review by yujun777 — APPROVED** [review]
LGTM

**Comment by deardeng** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/clone/DiskRebalancer.java [hunk 3]
```diff
@@ -185,7 +186,17 @@ protected List<TabletSchedCtx> selectAlternativeTabletsForCluster(
             Set<Long> pathHigh = Sets.newHashSet();
             // we only select tablets from available high load path
             beStat.getPathStatisticByClass(pathLow, pathMid, pathHigh, medium);
+            LOG.info("dx test select before low={} mid={} high={} medium={}", pathLow, pathMid, pathHigh, medium);
             // check if BE has low and high paths for balance after reclassification
+            pathHigh.add(-2606726262674133323L);
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/clone/DiskRebalancer.java:191]: test code

File: fe/fe-core/src/main/java/org/apache/doris/clone/DiskRebalancer.java [hunk 4]
```diff
@@ -120,6 +120,7 @@ private boolean checkAndReclassifyPaths(Set<Long> pathLow, Set<Long> pathMid, Se
     @Override
     protected List<TabletSchedCtx> selectAlternativeTabletsForCluster(
             LoadStatisticForTag clusterStat, TStorageMedium medium) {
+        LOG.info("dx test enter selectAlternativeTabletsForCluster");
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/clone/DiskRebalancer.java:123]: test code

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**